services/federator/test/integration/Test/Federator/InwardSpec.hs

```diff
@@ -72,6 +75,14 @@ spec env =
             <!! const 200 === statusCode
         liftIO $ bdy `shouldBe` expectedProfile
 
+    -- @SF.Federation @TSFI.RESTfulAPI @S2 @S3 @S7
+    --
+    -- (This is also tested in unit tests; search for
+    -- 'validateDomainCertInvalid' and 'testDiscoveryFailure'.)
+    it "shouldRejectMissmatchingOriginDomainInward" $
+      runTestFederator env $ pure ()
```

Comment:
```suggestion
      pure ()
```

Reply:
I did not look thoroughly as to why, but just `pure ()` doesn't type-check. I think that's because other `it` "blocks" in the `spec` use the `runTestFederator env test` values, which then GHC infers a type from, which then forces us to have the same type here, but I'd have to look deeper to confirm this.

Reply:
ah, you're almost right: `pure ()` can have the right type, it's an inference problem because both `it` and `pure` are polymorphic, and nothing determines any concrete type for `m`.  you could write:

```
it "..." $ (pure () :: Assertion)
```

this would be much better if `runTestFederator` would set up expensive schaffolding (which is what I thought earlier), but since it's just Reader+IO, I would say just leave it as is.